static/js/server_events_dispatch.js

```diff
@@ -389,15 +389,14 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
 
     case 'delete_message':
         var msg_id = event.message_id;
-        var message = message_store.get(msg_id);
         // message is passed to unread.get_unread_messages,
         // which returns all the unread messages out of a given list.
         // So double marking something as read would not occur
-        unread_ops.mark_message_as_read(message);
-        if (message.type === 'stream') {
+        unread_ops.process_read_messages_event([msg_id]);
```

Comment:
Make the `unread_ops` change its own commit separate from the rest, with a clear commit message explaining why the change is needed and correct.